raft/progress.go

```diff
@@ -203,6 +202,17 @@ func (in *inflights) add(inflight uint64) {
 	if next >= in.size {
 		next -= in.size
 	}
+	if next >= len(in.buffer) {
```

Comment:
Moved this into an `inflights.growBuf` method and added a comment. Let me know if the comment is sufficient or if you want additional commentary elsewhere.